common-modules/src/main/java/com/softserve/if072/common/model/ProductStatistics.java

```diff
@@ -0,0 +1,251 @@
+package com.softserve.if072.common.model;
+
+import java.sql.Timestamp;
+import java.util.Arrays;
+
+/**
+ * The ProductStatistics class stores information about the main statistics properties of products using
+ *
+ * @author Igor Kryviuk
+ */
+public class ProductStatistics {
+    private int productId;
+    private String productName;
+    private String productUnit;
+    private int amountInStorage;
+    private Timestamp lastUsingDate;
+    private Timestamp lastPurchasingDate;
+    private Timestamp endDate;
+    private double mean;
+    private double min;
+    private double max;
+    private double count;
+    private double range;
+    private int totalPurchased;
+    private int totalUsed;
+    private double standardDeviation;
+    private double lowerThreeSigmaLimit;
+    private double upperThreeSigmaLimit;
+    private double[] productUsingSpeeds;
+    private Timestamp[] purchasingProductDates;
+    private int[] purchasingProductAmounts;
+    private Timestamp[] usingProductDates;
+    private int[] usingProductAmounts;
+    private boolean hasOutliers;
+
+    public ProductStatistics() {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1186)